test/jdk/java/awt/print/PrinterJob/PrintAllFonts.java

```diff
@@ -21,55 +21,61 @@
  * questions.
  */
 
-/**
- *
- * test
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.GraphicsEnvironment;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+
+/*
+ * @test
  * @bug 4884389 7183516
  * @key printer
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary Font specified with face name loses style on printing
  * @run main/manual PrintAllFonts
  */
-
-import java.awt.*;
-import java.awt.print.*;
-import java.awt.GraphicsEnvironment;
-
 public class PrintAllFonts implements Printable {
+    private static final int LINE_HEIGHT = 18;
+    private static final int FONT_SIZE = 14;
 
-    static Font[] allFonts;
-    int fontNum = 0;
-    int startNum = 0;
-    int lineHeight = 18;
-    boolean done = false;
-    int thisPage = 0;
+    private final Font[] allFonts =
+            GraphicsEnvironment.getLocalGraphicsEnvironment().getAllFonts();
+    private int fontNum = 0;
+    private int startNum = 0;
+    private int thisPage = 0;
 
+    private static final String INSTRUCTIONS =
+            "This bug is system dependent and is not always reproducible.\n" +
+            "Font names will be printed in two columns.\n" +
+            "First column non synthesised and second column with synthesised italic.\n" +
+            "A passing test will have all text printed with correct font style.";
 
     public static void main(String[] args) throws Exception {
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new RuntimeException("Printer not configured or available.");
+        }
 
-        String[] instructions =
-        {
-            "You must have a printer available to perform this test and should use Win 98.",
-            "This bug is system dependent and is not always reproducible.",
-            " ",
-            "A passing test will have all text printed with correct font style.",
-        };
-
-        Sysout.createDialog( );
-        Sysout.printInstructions( instructions );
-
-        GraphicsEnvironment ge =
-            GraphicsEnvironment.getLocalGraphicsEnvironment();
-        allFonts = ge.getAllFonts();
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
```

Comment:
```suggestion
        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
```
You have to use the `builder()` method, otherwise the test will fail to compile after integration. See PR #18206 for [JDK-8325851](https://bugs.openjdk.org/browse/JDK-8325851).